src/Backups/BackupEntryWithChecksumCalculation.cpp

```diff
@@ -0,0 +1,54 @@
+#include <Backups/BackupEntryWithChecksumCalculation.h>
+#include <IO/HashingReadBuffer.h>
+
+
+namespace DB
+{
+
+template <typename Base>
+UInt128 BackupEntryWithChecksumCalculation<Base>::getChecksum() const
+{
+    std::lock_guard lock{checksum_calculation_mutex};
+    if (!calculated_checksum)
+    {
+        auto read_buffer = this->getReadBuffer(ReadSettings{}.adjustBufferSize(this->getSize()));
+        HashingReadBuffer hashing_read_buffer(*read_buffer);
+        hashing_read_buffer.ignoreAll();
+        calculated_checksum = hashing_read_buffer.getHash();
+    }
+    return *calculated_checksum;
+}
+
+template <typename Base>
+std::optional<UInt128> BackupEntryWithChecksumCalculation<Base>::getPartialChecksum(size_t prefix_length) const
+{
+    if (prefix_length == 0)
```

Comment:
According to the code which calls this method `prefix_length` couldn't be 0...

Reply:
Right, normally `prefix_length` must not be zero, but since it's very easy we can support zero here just for completeness.